### DISCUSSION TITLE
[Enhancement] [sub-menu] menu组件在窄边情况下显示more菜单时, more菜单无法设置popper-offset

### DISCUSSION THREAD

**Discussion by xxiaojiujiu** [element-plus/element-plus#14724] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `sub-menu`

## Description

我在窄屏的情况下显示了默认的"更多"菜单项，前面的菜单都可以设置popper-offset，但是最后一个“更多”菜单项无法设置它的popper-offset，这里有一个例子
https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLW1lbnVcbiAgICA6ZGVmYXVsdC1hY3RpdmU9XCJhY3RpdmVJbmRleDJcIlxuICAgIGNsYXNzPVwiZWwtbWVudS1kZW1vXCJcbiAgICBtb2RlPVwiaG9yaXpvbnRhbFwiXG4gICAgYmFja2dyb3VuZC1jb2xvcj1cIiM1NDVjNjRcIlxuICAgIHRleHQtY29sb3I9XCIjZmZmXCJcbiAgICBhY3RpdmUtdGV4dC1jb2xvcj1cIiNmZmQwNGJcIlxuICAgIEBzZWxlY3Q9XCJoYW5kbGVTZWxlY3RcIlxuICA+XG4gICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjFcIj5Qcm9jZXNzaW5nIENlbnRlcjwvZWwtbWVudS1pdGVtPlxuICAgIDxlbC1zdWItbWVudSBpbmRleD1cIjJcIiA6cG9wcGVyLW9mZnNldD1cIjMwXCI+XG4gICAgICA8dGVtcGxhdGUgI3RpdGxlPldvcmtzcGFjZTwvdGVtcGxhdGU+XG4gICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiMi0xXCI+aXRlbSBvbmU8L2VsLW1lbnUtaXRlbT5cbiAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIyLTJcIj5pdGVtIHR3bzwvZWwtbWVudS1pdGVtPlxuICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjItM1wiPml0ZW0gdGhyZWU8L2VsLW1lbnUtaXRlbT5cbiAgICAgIDxlbC1zdWItbWVudSBpbmRleD1cIjItNFwiPlxuICAgICAgICA8dGVtcGxhdGUgI3RpdGxlPml0ZW0gZm91cjwvdGVtcGxhdGU+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIyLTQtMVwiPml0ZW0gb25lPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIyLTQtMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCIyLTQtM1wiPml0ZW0gdGhyZWU8L2VsLW1lbnUtaXRlbT5cbiAgICAgIDwvZWwtc3ViLW1lbnU+XG4gICAgPC9lbC1zdWItbWVudT5cbiAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiM1wiIGRpc2FibGVkPkluZm88L2VsLW1lbnUtaXRlbT5cbiAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiNFwiPk9yZGVyczwvZWwtbWVudS1pdGVtPlxuICAgIDxlbC1zdWItbWVudSBpbmRleD1cIjVcIiA6cG9wcGVyLW9mZnNldD1cIjMwXCI+XG4gICAgICA8dGVtcGxhdGUgI3RpdGxlPldvcmtzcGFjZTwvdGVtcGxhdGU+XG4gICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiNS0xXCI+aXRlbSBvbmU8L2VsLW1lbnUtaXRlbT5cbiAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCI1LTJcIj5pdGVtIHR3bzwvZWwtbWVudS1pdGVtPlxuICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjUtM1wiPml0ZW0gdGhyZWU8L2VsLW1lbnUtaXRlbT5cbiAgICAgIDxlbC1zdWItbWVudSBpbmRleD1cIjUtNFwiPlxuICAgICAgICA8dGVtcGxhdGUgI3RpdGxlPml0ZW0gZm91cjwvdGVtcGxhdGU+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCI1LTQtMVwiPml0ZW0gb25lPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCI1LTQtMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCI1LTQtM1wiPml0ZW0gdGhyZWU8L2VsLW1lbnUtaXRlbT5cbiAgICAgIDwvZWwtc3ViLW1lbnU+XG4gICAgPC9lbC1zdWItbWVudT5cbiAgICA8ZWwtc3ViLW1lbnUgaW5kZXg9XCI2XCIgOnBvcHBlci1vZmZzZXQ9XCIzMFwiPlxuICAgICAgPHRlbXBsYXRlICN0aXRsZT5Xb3Jrc3BhY2U8L3RlbXBsYXRlPlxuICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjYtMVwiPml0ZW0gb25lPC9lbC1tZW51LWl0ZW0+XG4gICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiNi0yXCI+aXRlbSB0d288L2VsLW1lbnUtaXRlbT5cbiAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCI2LTNcIj5pdGVtIHRocmVlPC9lbC1tZW51LWl0ZW0+XG4gICAgICA8ZWwtc3ViLW1lbnUgaW5kZXg9XCI2LTRcIj5cbiAgICAgICAgPHRlbXBsYXRlICN0aXRsZT5pdGVtIGZvdXI8L3RlbXBsYXRlPlxuICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiNi00LTFcIj5pdGVtIG9uZTwvZWwtbWVudS1pdGVtPlxuICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiNi00LTJcIj5pdGVtIHR3bzwvZWwtbWVudS1pdGVtPlxuICAgICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiNi00LTNcIj5pdGVtIHRocmVlPC9lbC1tZW51LWl0ZW0+XG4gICAgICA8L2VsLXN1Yi1tZW51PlxuICAgIDwvZWwtc3ViLW1lbnU+XG4gICAgPGVsLXN1Yi1tZW51IGluZGV4PVwiN1wiIDpwb3BwZXItb2Zmc2V0PVwiMzBcIj5cbiAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+V29ya3NwYWNlPC90ZW1wbGF0ZT5cbiAgICAgIDxlbC1tZW51LWl0ZW0gaW5kZXg9XCI3LTFcIj5pdGVtIG9uZTwvZWwtbWVudS1pdGVtPlxuICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjctMlwiPml0ZW0gdHdvPC9lbC1tZW51LWl0ZW0+XG4gICAgICA8ZWwtbWVudS1pdGVtIGluZGV4PVwiNy0zXCI+aXRlbSB0aHJlZTwvZWwtbWVudS1pdGVtPlxuICAgICAgPGVsLXN1Yi1tZW51IGluZGV4PVwiNy00XCI+XG4gICAgICAgIDx0ZW1wbGF0ZSAjdGl0bGU+aXRlbSBmb3VyPC90ZW1wbGF0ZT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjctNC0xXCI+aXRlbSBvbmU8L2VsLW1lbnUtaXRlbT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjctNC0yXCI+aXRlbSB0d288L2VsLW1lbnUtaXRlbT5cbiAgICAgICAgPGVsLW1lbnUtaXRlbSBpbmRleD1cIjctNC0zXCI+aXRlbSB0aHJlZTwvZWwtbWVudS1pdGVtPlxuICAgICAgPC9lbC1zdWItbWVudT5cbiAgICA8L2VsLXN1Yi1tZW51PlxuICA8L2VsLW1lbnU+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCBhY3RpdmVJbmRleCA9IHJlZignMScpXG5jb25zdCBhY3RpdmVJbmRleDIgPSByZWYoJzEnKVxuY29uc3QgaGFuZGxlU2VsZWN0ID0gKGtleTogc3RyaW5nLCBrZXlQYXRoOiBzdHJpbmdbXSkgPT4ge1xuICBjb25zb2xlLmxvZyhrZXksIGtleVBhdGgpXG59XG48L3NjcmlwdD5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIl9vIjp7fX0=

### Additional comments

可以新增一个属性来控制

![image](https://github.com/element-plus/element-plus/assets/35479120/fd692d80-edec-498a-b8d3-f9eb9716ae5f)
![image](https://github.com/element-plus/element-plus/assets/35479120/9d187a05-e939-4ffa-9a13-8921d79f5b1f)

**Comment by kooriookami** [comment]
#15051
